File: Lab4/pzpi-22-3-bilozub-danylo-lab4/ZooShopAndroid/ZooShopAndroid/app/src/main/java/com/example/zooshop/model/Sensor.kt
package com.example.zooshop.model
data class Sensor(
    val sensor_id: Int,
    val type: String,
    val location: String,
    val current_value: Float,
    val last_maintenance: String,
    val sensor_link: String?
)
data class SensorCreate(
    val shop_id: Int,
    val type: String,
    val location: String,
    val sensor_link: String? = null
)

File: Lab3/pzpi-22-3-bilozub-danylo-lab3/ZooShopHelper/app/schemas/sensor.py
from datetime import datetime
from pydantic import BaseModel

class SensorBase(BaseModel):
    type: str
    location: str
    current_value: float
    last_maintenance: datetime
    shop_id: int
    sensor_link: str
    model_config = {
        "from_attributes": True
    }

class SensorCreate(SensorBase):
    pass

class SensorUpdate(SensorBase):
    pass

class Sensor1(SensorBase):
    sensor_id: int

    model_config = {
        "from_attributes": True
    }
